File: web/src/app/classes/order.ts
import { ShippingDetails } from './shipping-details';
import { Cart } from './cart';
import { OrderReason } from './order-reason';
import { FileViewer } from './file-viewer';
import { OrderEmailStatus } from './order-email-status';
import { Product } from './product';

export class Order {

    id: number;
    orderNumber: number;
    dateOrder: Date;
    customerName: string;
    email: string;
    amount: number;
    shippingId: number;
    paymentStatusId: number;
    paymentStatus: string;
    layaway: boolean;
    statusId: number;
    status: string;
    invoiceNumber: number;
    shippingAddress: string;
    shippingDetails: ShippingDetails;
    paymentMethodDescription: string;
    paymentMethodName: string;
    orderCart: Array<Cart>;
    trackingNumber: string;
    shippingDate: Date;
    ipAddress: string;
    withPreOrder: boolean;
    transactionFee: number;
    finalAmount: number;
    insuranceFee: number;
    isEmailSubscribe: boolean;
    // for email
    amountToPay: number;
    discountAmount: number;
    shippingAmount: number;
    totalPrice: number;
    orderReason: OrderReason;
    paynowEnable: boolean;
    reason: string;
    orderAttachment: Array<FileViewer>;
    orderEmailStatus: Array<OrderEmailStatus>;
    securityId: string;
    forLayAway: boolean;
    layAwayId: number;
    isSendEmail: boolean;
    trackingUrl: string;
    forPreOrder: boolean;
    preOrderId: number;
    isPrSend: boolean;
    forPr: boolean;
    preOrderLayaway: boolean;
    paymongoStatus: number;
    NotifProduct: Product;
    forPreOrderNotif: boolean;

    constructor() {
        this.orderCart = new Array<Cart>();
        this.paymentMethodName = '';
        this.orderReason = new OrderReason();
        this.shippingDetails = new ShippingDetails();
        this.orderAttachment = new Array<FileViewer>();
        this.orderEmailStatus = new Array<OrderEmailStatus>();
        this.NotifProduct = new Product();
        this.trackingNumber = '';
        this.forPr = false;
        this.layaway = false;
        this.withPreOrder = false;
    }

}

File: api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ByzmoApi.Models
{
    public class Order
    {
        public int Id { get; set; }
        public int OrderNumber { get; set; }
        public string InvoiceNumber { get; set; }
        public DateTimeOffset DateOrder { get; set; }
        public int ShippingId { get; set; }
        public int PaymentStatusId { get; set; }
        public int StatusId { get; set; }
        public string CustomerName { get; set; }
        public string Email { get; set; }
        public decimal Amount { get; set; }
        public bool LayAway { get; set; }
        public string ShippingAddress { get; set; }
        public ShippingDetails ShippingDetails { get; set; }
        public string PaymentMethodDescription { get; set; }
        public string PaymentMethodName { get; set; }
        public IEnumerable<Cart> OrderCart { get; set; }
        public string TrackingNumber { get; set; }
        public DateTimeOffset ShippingDate { get; set; }
        public string IpAddress { get; set; }
        public bool WithPreOrder { get; set; }
        public OrderReason OrderReason { get; set; }
        public string Reason { get; set; }
        public IEnumerable<OrderAttachment> OrderAttachment { get; set; }
        public IEnumerable<OrderEmailStatus> OrderEmailStatus { get; set; }
        public string SecurityId { get; set; }
        public bool IsEmailSubscribe { get; set; }
        public int PaymongoStatus  { get; set; }
        //for email
        public decimal AmountToPay { get; set; }
        public decimal DiscountAmount { get; set; }
        public decimal ShippingAmount { get; set; }
        public decimal InsuranceFee { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal TransactionFee { get; set; }
        public decimal FinalAmount { get; set; }

        public bool ForLayAway { get; set; }
        public int LayAwayId { get; set; }
        public bool IsSendEmail { get; set; }
        public bool ForPreOrder { get; set; }
        public int PreOrderId { get; set; }
        public bool IsPrSend { get; set; }
        public bool ForPr { get; set; }
        public IEnumerable<OrderEmailHeader> OrderEmailHeader { get; set; }
        public bool ForPreOrderNotif { get; set; }
        public Product NotifProduct { get; set; }
    }
}